multinet/api.py

```diff
@@ -24,11 +27,62 @@
     AlreadyExists,
     RequiredParamsMissing,
 )
-from multinet.user import current_user
+from multinet.user import current_user, find_user_from_id
 
 bp = Blueprint("multinet", __name__)
 
 
+# Included here due to circular imports
+# TODO: Remove once implementing new ORM and permission storage
+def _permissions_id_to_user(permissons: WorkspacePermissions) -> Dict:
+    """Transform permission documents to directly container user info."""
+
+    new_permissions = deepcopy(permissons)
+    for role, users in new_permissions.items():
+        if role == "public":
+            continue
+
+        if role == "owner":
+            assert isinstance(users, str)
+
+            user = find_user_from_id(users)
+            if user is not None:
+                # Not sure why mypy complains about `user` here
+                new_permissions["owner"] = asdict(user)  # type: ignore
+        else:
+            assert isinstance(users, list)
+
+            new_users = []
+            for sub in users:
+                user = find_user_from_id(sub)
```

Comment:
Does this make the loop O(N^2)? Is that ok?

Reply:
I don't think it's a problem, since this it's really O(n*m), where n is the number of keys in the permissions dict (5), and m is the number of users in each role (could just use the role with the most users, to be safe). So it's really linear.